ports/esp32/modmachine.c

```diff
@@ -101,11 +100,11 @@ static mp_obj_t mp_machine_get_freq(void) {
 static void mp_machine_set_freq(size_t n_args, const mp_obj_t *args) {
     mp_int_t freq = mp_obj_get_int(args[0]) / 1000000;
     if (freq != 20 && freq != 40 && freq != 80 && freq != 160
-        #if !CONFIG_IDF_TARGET_ESP32C3
+        #if !(CONFIG_IDF_TARGET_ESP32C3 || CONFIG_IDF_TARGET_ESP32C6)
         && freq != 240
         #endif
         ) {
-        #if CONFIG_IDF_TARGET_ESP32C3
+        #if CONFIG_IDF_TARGET_ESP32C3 || CONFIG_IDF_TARGET_ESP32C6
         mp_raise_ValueError(MP_ERROR_TEXT("frequency must be 20MHz, 40MHz, 80Mhz or 160MHz"));
         #else
         mp_raise_ValueError(MP_ERROR_TEXT("frequency must be 20MHz, 40MHz, 80Mhz, 160MHz or 240MHz"));
```

Comment:
@projectgus I went hunting for a better way to handle this but came up with nothing. The individual values seem to be either hardcoded in soc specific c files or calculated from available dividers. eg. https://github.com/espressif/esp-idf/blob/6673376297b921d438790d195014b860eaf8bb17/components/esp_hw_support/port/esp32c2/rtc_clk.c#L174

This does suggest this block will get much worse when bringing in newer chips though: https://github.com/espressif/esp-idf/blob/6673376297b921d438790d195014b860eaf8bb17/docs/en/api-reference/system/power_management.rst#power-management-locks